### ISSUE TITLE
PRs should build documentation, too

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#1811]
The build servers are not actually building doxygen or sphinx documentation.  Bad syntax in a commit could bring down our hosted documentation (which is now our front page).

**Comment by jamiesnape** [comment]
You should also set the documentation to update during a successful nightly build. This also minimizes your risks.

**Comment by sherm1** [comment]
I would like to be able to test the doxygen I write along with code changes in my local build before I commit them. It would be nice to have a build target to allow `make doxygen` or building docs from within the IDE (is this worthy of a separate issue?). Meanwhile, is there an easy way to build doxygen locally (and apologies if I missed obvious instructions somewhere)?

**Comment by RussTedrake** [comment]
`make doc` in drake-distro/drake 
it was the logical place for it when drake was a module.  now feels illogical in the superbuild format.  and would be great to push it into cmake.

**Comment by sherm1** [comment]
Opened issue #1870 requesting a doxygen target.

**Comment by jamiesnape** [comment]
Depends on #2143.
